config/vhosts/vhosts.inc

```diff
@@ -65,49 +82,45 @@ function vhosts_sync_package() {
 	global $config;
 
 	if ($config['installedpackages']['vhosts']['config'] != "") {
-		conf_mount_rw();
+		// # NOTE not needed anymore
+//		conf_mount_rw();
 
-		//sort the vhosts array
 		$vhostarray = $config['installedpackages']['vhosts']['config'];
-		if (count(vhostarray) > 1) {
-			usort($vhostarray, 'sort_ipaddress');
-			usort($vhostarray, 'sort_host');
-			usort($vhostarray, 'sort_port');
-		}
+		// # TODO remove loop to acquire enabled vhosts
+		// # NOTE same work can be done within loop to build conf file
 		$vhostarray_http = '';
-		$vhostarray_https = '';
+		// # HACK setting the Default port
+		$server_port = '8000';
 		$x = 0;
 		foreach ($vhostarray as $rowhelper) {
 			if ($rowhelper['enabled'] != "false") {
-				if (strlen($rowhelper['certificate']) > 0 && strlen($rowhelper['privatekey']) > 0) {
-					$vhostarray_https[$x]['host'] = $rowhelper['host'];
-					$vhostarray_https[$x]['ipaddress'] = $rowhelper['ipaddress'];
-					$vhostarray_https[$x]['port'] = $rowhelper['port'];
-					$vhostarray_https[$x]['directory'] = $rowhelper['directory'];
-					$vhostarray_https[$x]['certificate'] = $rowhelper['certificate'];
-					$vhostarray_https[$x]['privatekey'] = $rowhelper['privatekey'];
-					$vhostarray_https[$x]['enabled'] = $rowhelper['enabled'];
-					$vhostarray_https[$x]['description'] = $rowhelper['description'];
-				} else {
 					$vhostarray_http[$x]['host'] = $rowhelper['host'];
 					$vhostarray_http[$x]['ipaddress'] = $rowhelper['ipaddress'];
-					$vhostarray_http[$x]['port'] = $rowhelper['port'];
+					$vhostarray_http[$x]['port'] = (strlen($rowhelper['port']) == 0) ? '8001' : $rowhelper['port']; // # XXX do not need to set the default port
```

Comment:
Yeah indeed... Things like 80 / 443 / ```$config['system']['webgui]['port']``` / 3128+8080 / 8000~8100 should be caught by input validation and not allowed to be set up at all.

@dariomas - let me know when the code is somehow finished otherwise and I can add some input validations as pull request here. (No point if doing so until you are done here, though.)